app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -29,12 +33,35 @@
     // enum values with DEFAULT actions //
     //////////////////////////////////////
 
-    show_channel_details(R.string.show_channel_details, (fragment, item) ->
-        // For some reason `getParentFragmentManager()` doesn't work, but this does.
-        NavigationHelper.openChannelFragment(
-                fragment.requireActivity().getSupportFragmentManager(),
-                item.getServiceId(), item.getUploaderUrl(), item.getUploaderName())
-    ),
+    show_channel_details(R.string.show_channel_details, (fragment, item) -> {
+        if (isNullOrEmpty(item.getUploaderUrl())) {
+            final int serviceId = item.getServiceId();
+            final String url = item.getUrl();
+            // TODO: Some visual loading indicator
+            ExtractorHelper.getStreamInfo(serviceId, url, false)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(result -> {
+                        NewPipeDatabase.getInstance(fragment.getContext()).streamDAO()
+                                .setUploaderUrl(serviceId, url, result.getUploaderUrl())
+                                .subscribeOn(Schedulers.io()).subscribe();
+                        // For some reason `getParentFragmentManager()` doesn't work, but this does.
+                        NavigationHelper.openChannelFragment(
+                                fragment.requireActivity().getSupportFragmentManager(),
+                                item.getServiceId(), result.getUploaderUrl(),
+                                item.getUploaderName());
```

Comment:
This block is duplicated. see line 60-64

Reply:
Fixed in https://github.com/TeamNewPipe/NewPipe/pull/6919/commits/bd0427c79f4474bc2c3194672d11d8223f7d4520